flipperdevices/flipperzero-firmware
# Run Bad USB immediately after connection

# What's new

- The `Run` button in Bad USB now appears before connection. If pressed at that time, the screen will display `Will run on connect`. After that, when Flipper is connected by USB, the script's execution will start after a 1.5s delay (which gives the PC proper time to recognize Flipper as a keyboard).
- The button's name will also change into `Cancel` after you press it. If pressed again, the scheduled execution will cancel, and Flipper will return to the normal pre-connection mode.
- Solves #1903.

# Verification 

- Disconnect Flipper from USB
- Open Bad USB app and select a script to run
- Press `Run`. The on-screen text should change to `Will run on connect`
- Connect Flipper to USB. The selected script should instantly run
- Disconnect Flipper from USB
- Press `Run` and then press `Cancel`. The on-screen text should change and then change back to `Connect to USB`
- Connect Flipper to USB. Normal behavior should ensue, i.e. the script should not run automatically

# Checklist (For Reviewer)
